simplesat/transaction.py

```diff
@@ -113,23 +121,40 @@ def _safe_operations(self, pool, decisions, installed_package_ids):
                       package_id not in installed_package_ids):
                     installs.append(package_id)
 
-        # Removals should happen top down
-        for package_id in reversed(removals):
-            package = pool.id_to_package(package_id)
-            operations.append(RemoveOperation(package))
+        install_operations = []
+        remove_operations = []
 
         # Installations should happen bottom up
         for package_id in installs:
-            package = pool.id_to_package(package_id)
-            operations.append(InstallOperation(package))
+            # The solver may suggest removing a currently installed package and
+            # installing another version of that same package from a different
```

Comment:
As I currently understand it, we pass around the repo information but we don't actually do anything with it. The only place where it matters is when considering which packages are currently installed (available in the installed repo). This would be relevant if we implemented repo prioritization. Can you think of any other situations when we would care? Specifically, what would it mean if those identical-looking packages from different repositories are not actually equivalent?